pkg/util/pki/kube.go

```diff
@@ -52,6 +46,26 @@ func GenerateTemplateFromCertificateSigningRequest(csr *certificatesv1.Certifica
 	return GenerateTemplateFromCSRPEMWithUsages(csr.Spec.Request, duration, isCA, ku, eku)
 }
 
+// DurationFromCertificateSigningRequest will return the time.Duration of the
+// requested duration on the CertificateSigningRequest. If the annotation is
+// empty, will return the cert-manager default certificate duration
+func DurationFromCertificateSigningRequest(csr *certificatesv1.CertificateSigningRequest) (time.Duration, error) {
```

Comment:
**nit:** long names 😥

Would it make sense to keep it short and rename it to either `DurationFromCSR` or `DurationFrom`?

**nit 2:** dot at the end of the comment, use present tense

```suggestion
// DurationFrom returns the duration that the user may have requested using the
// annotation "experimental.cert-manager.io/request-duration". Returns the
// cert-manager default certificate duration when the user hasn't provided the
// annotation.
func DurationFromCertificateSigningRequest(csr *certificatesv1.CertificateSigningRequest) (time.Duration, error) {
```

Reply:
We use `CertificateSigningRequest` here throughout as to not cause conflicts or confusion with `x509.CertificateRequest` which currently occupies `CSR` in most of the code base, including here.

Reply:
Will update the comment.

Reply:
Yeah, you are right, "csr" and "CSR" are both already taken by `x509.CertificateRequest`. Thanks for pointing that out!